src/index.ts

```diff
@@ -142,33 +174,177 @@ export class EmbeddingIndex {
+    const useStorage = options.useStorage || 'none';
 
-    // Compute similarities
-    const similarities = this.objects
-      .filter((object) =>
-        Object.keys(filter).every((key) => object[key] === filter[key])
-      )
-      .map((obj) => ({
-        similarity: cosineSimilarity(queryEmbedding, obj.embedding),
-        object: obj,
-      }));
+    if (useStorage === 'indexedDB') {
+      const DBname = options.storageOptions?.indexedDBName || 'clientVectorDB';
+      const objectStoreName =
+        options.storageOptions?.indexedDBObjectStoreName ||
+        'ClientEmbeddingStore';
 
-    // Sort by similarity and return topK results
-    return similarities
-      .sort((a, b) => b.similarity - a.similarity)
-      .slice(0, topK);
+      if (typeof indexedDB === 'undefined') {
+        console.error('IndexedDB is not supported');
+        throw new Error('IndexedDB is not supported');
+      }
+      const results = await this.loadAndSearchFromIndexedDB(
+        DBname,
+        objectStoreName,
+        queryEmbedding,
+        topK,
+        filter,
+      );
+      return results;
+    } else {
+      // Compute similarities
+      const similarities = this.objects
+        .filter((object) =>
+          Object.keys(filter).every((key) => object[key] === filter[key]),
+        )
+        .map((obj) => ({
+          similarity: cosineSimilarity(queryEmbedding, obj.embedding),
+          object: obj,
+        }));
+
+      // Sort by similarity and return topK results
+      return similarities
+        .sort((a, b) => b.similarity - a.similarity)
+        .slice(0, topK);
+    }
   }
 
   printIndex() {
-    console.log("Index Content:");
+    console.log('Index Content:');
     this.objects.forEach((obj, idx) => {
       console.log(`Item ${idx + 1}:`, obj);
     });
   }
+
+  async saveIndex(
+    storageType: string,
+    options: { DBName: string; objectStoreName: string } = {
+      DBName: 'clientVectorDB',
+      objectStoreName: 'ClientEmbeddingStore',
+    },
+  ) {
+    if (storageType === 'indexedDB') {
+      await this.saveToIndexedDB(options.DBName, options.objectStoreName);
+    } else {
+      throw new Error(
+        `Unsupported storage type: ${storageType} \n Supported storage types: "indexedDB"`,
+      );
+    }
+  }
+
+  async saveToIndexedDB(
+    DBname: string = 'clientVectorDB',
+    objectStoreName: string = 'ClientEmbeddingStore',
+  ): Promise<void> {
+    if (typeof indexedDB === 'undefined') {
+      console.error('IndexedDB is not defined');
+      throw new Error('IndexedDB is not supported');
+    }
+    return new Promise(async (resolve, reject) => {
+      if (!this.objects || this.objects.length === 0) {
+        reject(new Error('Index is empty. Nothing to save'));
+        return;
+      }
+
+      const db = await IndexedDbManager.create(DBname, objectStoreName);
+
+      await db
+        .addToIndexedDB(this.objects)
+        .then(() => {
+          console.log(
```

Comment:
We probably dont want to log console anything unless its super critical or an error.